### DISCUSSION TITLE
Do not delete the associated data flow when a component is deleted?

### DISCUSSION THREAD

**Discussion by jgadsden** [OWASP/threat-dragon#640] [Q&A] [answered]
When a component is deleted off the threat model diagram then all connected data flows are also deleted

This has been reported as a problem by various users. It may be better not to delete the data flows?

Should the behaviour change to preserve the data flows when the component is deleted?

**Comment by srcnuzn** [comment, marked as answer]
I agree. Or at least #1069 should be fixed. I lost work due to this behavior.

**Comment by jgadsden** [comment]
OK, we go with the recommendation in #1069 , thanks @srcnuzn
